metis/lib/client/jsx/components/folder-view.jsx

```diff
@@ -35,10 +50,30 @@ const FolderView = ({bucket_name, folder_name}) => {
   const uploadFileInput = useRef(null);
   const uploadDirInput = useRef(null);
 
+  const [loading, setLoading] = useState(true)
+  const [showLoading, setShowLoading] = useState(false)
+
+  const classes = useStyles();
+
   useEffect(() => {
+    setLoading(true);
     invoke({type: 'RETRIEVE_FILES', bucket_name, folder_name});
+    setLoading(false);
   }, []);
 
+  useAsync( async () => {
+    if (loading && !showLoading) {
+      const delay = (delayInms) => {
+        return new Promise(resolve => setTimeout(resolve, delayInms));
+      };
+      let delayres = await delay(100);
+      setShowLoading(true);
+    }
+    if (!loading) {
+      setShowLoading(false);
+    }
+  }, [loading, showLoading]);
```

Comment:
Why is `showLoading` a dependency? I feel like you'd only want this fn to be called when `loading` changes.
```suggestion
  }, [loading, showLoading]);
```

Reply:
That's fair.  I agree it doesn't need to be a dependency.  The linter in our testing setup will complain, but only with a warning, so I can remove.